.github/workflows/test-hyperlinks.md

```diff
@@ -0,0 +1,274 @@
+```
+### Test External links
+The external links are tested using `curl`  command. The `-I` tag helps in fetching only the HTTP-request header. The `-L` flag helps in redirecting to the new location specified in the Location header. After fetching the status code of each link, the link and its status code is uploaded into the `checkedLinksCache.txt`, a file that helps in caching so that redundant links are tested only once. So before checking the status code through `curl` command. We check the `checkedLinksCache.txt` file if the link exists, if so then the status code is fetched from the cache file. If the status code of the link is found to be 404, then the link is uploaded to the `error-report.log` log file.
+```yaml    
+- name: Test External links
+  run: |
+    touch checkedLinksCache.txt
+    IFS=$'\n'
+    for LINE in $(cat externalLinks.txt )
+    do 
+      LINK=$( echo $LINE | cut -d ':' -f3- )
+      LINK=${LINK%.} #removing trailing .
+      LINK=${LINK% } #removing trailing space
+      CHECK_CACHE=$( cat checkedLinksCache.txt | grep "$LINK~" | wc -w )
+      TRY=3
+      if [[ $CHECK_CACHE -eq 0  ]]; then
+        while [[ $TRY -ne 0 ]]
+        do
+          STATUS_CODE=$(curl -LI -m 60 -s $LINK | grep "HTTP" | tail -1 | cut -d' ' -f2 ) 
+          if [[ -n $STATUS_CODE ]]; then
+            echo "$LINK~$STATUS_CODE" >> checkedLinksCache.txt
+            break
+          else  
+            echo $LINE
+            echo "retrying..."
+            TRY=$(( TRY - 1))
+          fi
+        done
+        else
+          STATUS_CODE=$( cat checkedLinksCache.txt | grep "$LINK~" | cut -d '~' -f2 )
+      fi
+      if [[ $STATUS_CODE -eq 404 ]]; then 
+        echo -e "${LINK} - \u001b[31m404 Error"
+        echo "${LINE}" >> error-report.log
+      elif [[ ! -n $STATUS_CODE ]]; then
+              echo -e "${LINK} - \033[0;31mNo Response\033[0m"
+              echo "${LINE}(No-Response)" >> error-report.log
+      else
+        echo "${LINK} - ${STATUS_CODE}" 
+      fi
+    done
+```
+### Test Internal Links
+Internal links are those that point to another file in the same repository or a reference to a section in the same file. For file references, it checks whether the mentioned file exists in the specified file reference path. For internal file section references, it checks if the specified reference section is present in the documentation.
+```yaml
+- name: Test Internal Links
+  run: |
+    for LINE in $( cat internalLinks.txt )
+    do 
+      REFERENCE=$( echo $LINE | cut -d ':' -f3- )
+      FILE=$( echo $LINE | cut -d ':' -f1 )
+      if [[ ${REFERENCE:0:1} == '#' ]]; then 
+        Link_text=$( cat $FILE | grep  -oE "\[.*\]\(${REFERENCE}\)" | sed 's/\[//' | cut -d ']' -f1 )
+        IS_PRESENT=$(cat $FILE | grep -oE "# ${Link_text}" | wc -w)
+        if [[ $IS_PRESENT -eq 0 ]]; then 
+				  echo -e "${LINE} -\u001b[31m invalid reference" 
```

Comment:
Not sure what the red bars are here and on line 204. They show on the rendered page as well.

Reply:
Removed those red bars.